Add SEO plugin

xstoicunicornx opened the pull request:
Add SEO plugin so that links in twitter and other social media sites look nice.

Here are the before and after open graph values from testing locally
<img width="774" height="539" alt="image" src="https://github.com/user-attachments/assets/f1bf7c9e-bcfc-4dd3-bc70-ad5095c990df" />
<img width="574" height="545" alt="image" src="https://github.com/user-attachments/assets/8dc324dd-b465-4cd1-99d9-0036e75f1130" />

xstoicunicornx commented:
Yes you are right, technically it says it needs to be a dev dependency. 

https://github.com/lorisleiva/vuepress-plugin-seo

What are you referring to when you say "debug pages"? Happy to try to take a look.

DanGould commented:
https://developers.facebook.com/tools/debug/ thanks for the help! Linkedin and twitter also have similar debuggers.

DanGould commented:
I think it'd need to be a full dep and not a dev dep to be used in production as well.

xstoicunicornx committed:
Fix SEO

xstoicunicornx committed:
Update card

xstoicunicornx commented:
ok! I made a lot of progress with some caveats.

first it looks like commit b788b7a47ca2e7bc2b98e7ba6531e7cb270702d7 was what originally broke SEO tags because the plugins in the spiral theme were being overridden, and the spiral theme is what had the `vuepress-plugin-seo` config to return the right meta tags.

so I have reverted those changes and added the card.png file that you were using in the `open-graph` branch you were working on. you can see it working at my [github page](https://xstoicunicornx.github.io). I tested it in a few different debug tools and seems to be working everywhere. (note, there is an extra commit in my xstoicunicornx/xstoicunicorn.github.io repo to change the baseurl but the second to last commit is 16f952d20c66c921c0e3e18013949b7e0a1636c4 same as this branch)

the caveat is that removing the plugin section will change the blog link structure. currently the blog links look like:

```https://payjoindevkit.org/2025/03/18/the-evolution-of-payjoin/```

but if this gets merged it will instead look like 

```https://payjoindevkit.org/blog/the-evolution-of-payjoin/```

that was the main affect I could identify from this change, however I am extremely unfamiliar with vuepress so there may be other side effects that I haven't been able to identify.

let me what you think :)
